MSCL_Unit_Tests/Test_InertialNode.cpp

```diff
@@ -101,6 +101,551 @@ BOOST_AUTO_TEST_CASE(InertialNode_getDataPackets_success)
+    Connection conn(connImpl);
+
+    //create the InertialNode object
+    InertialNode node(conn);
+
+    // enable NMEA parsing
+    node.enableNmeaParsing();
+
+    //check that there is no data to get
+    BOOST_CHECK_EQUAL(node.getNmeaPackets().size(), 0);
+}
+
+BOOST_AUTO_TEST_CASE(InertialNode_getNmeaPackets_success)
+{
+    std::shared_ptr<mockConnectionImpl> connImpl(new mockConnectionImpl);
+    Connection conn(connImpl);
+
+    //create the InertialNode object
+    InertialNode node(conn);
+
+    // enable NMEA parsing
+    node.enableNmeaParsing();
+
+    ByteStream bytes;
+
+    std::string pkt = "$PPKRR,154143.08,  nan,  nan,  nan,0.000000,0.000000,0.000000*38";
+    for (char c : pkt)
+    {
+        bytes.append_uint8(c);
+    }
+
+    bytes.append_uint16(0x0D0A);        //end of packet
+
+    pkt = "$GPGSV,3,1,9,2,23,165,44,5,18,103,44,8,3,335,0,13,26,52,37*46";
+    for (char c : pkt)
+    {
+        bytes.append_uint8(c);
+    }
+
+    bytes.append_uint16(0x0D0A);        //end of packet
+
+    connImpl->setResponseBytes(bytes);
+
+    //force parsing of the bytes we just set
+    connImpl->parseNextResponse();
+
+    // no MIP data found
+    BOOST_CHECK_EQUAL(node.totalPackets(), 0);
+
+    NmeaPackets pkts = node.getNmeaPackets();
+    BOOST_CHECK_EQUAL(pkts.size(), 2);
+    BOOST_CHECK_EQUAL(pkts.at(0).talkerId(), "P");
+    BOOST_CHECK_EQUAL(pkts.at(0).sentenceType(), "PKRR");
+    BOOST_CHECK_EQUAL(pkts.at(1).talkerId(), "GP");
+    BOOST_CHECK_EQUAL(pkts.at(1).sentenceType(), "GSV");
+}
+
+BOOST_AUTO_TEST_CASE(InertialNode_getNmeaPackets_valid_parserDisabled)
+{
+    std::shared_ptr<mockConnectionImpl> connImpl(new mockConnectionImpl);
+    Connection conn(connImpl);
+
+    //create the InertialNode object
+    InertialNode node(conn);
+
+    ByteStream bytes;
+
+    std::string pkt = "$PPKRR,154143.08,  nan,  nan,  nan,0.000000,0.000000,0.000000*38";
+    for (char c : pkt)
+    {
+        bytes.append_uint8(c);
+    }
+
+    bytes.append_uint16(0x0D0A);        //end of packet
+
+    pkt = "$GPGSV,3,1,9,2,23,165,44,5,18,103,44,8,3,335,0,13,26,52,37*46";
+    for (char c : pkt)
+    {
+        bytes.append_uint8(c);
+    }
+
+    bytes.append_uint16(0x0D0A);        //end of packet
+
+    connImpl->setResponseBytes(bytes);
+
+    //force parsing of the bytes we just set
+    connImpl->parseNextResponse();
+
+    // no MIP data found
+    BOOST_CHECK_EQUAL(node.totalPackets(), 0);
+
+    NmeaPackets pkts = node.getNmeaPackets();
+    BOOST_CHECK_EQUAL(pkts.size(), 0);
+}
+
+BOOST_AUTO_TEST_CASE(InertialNode_getNmeaPackets_success_wInvalidPackets)
+{
+    std::shared_ptr<mockConnectionImpl> connImpl(new mockConnectionImpl);
+    Connection conn(connImpl);
+
+    //create the InertialNode object
+    InertialNode node(conn);
+
+    // enable NMEA parsing
+    node.enableNmeaParsing();
+
+    ByteStream bytes;
+
+    std::string pkt = "$PPKRR,154143.08,  nan,  nan,  nan,0.000000,0.000000,0.000000*38";
```

Comment:
I assume this is a valid packet, but I would put a comment here saying that like you did for the invalid packets